astropy/io/ascii/core.py

```diff
@@ -671,8 +671,7 @@ class BaseData(object):
     write_spacer_lines = ['ASCII_TABLE_WRITE_SPACER_LINE']
     fill_include_names = None
     fill_exclude_names = None
-    # Currently, the default matches the numpy default for masked values.
-    fill_values = [(masked, '--')]
+    fill_values = [(masked, '')]
```

Comment:
I would add a comment in here: `# Note that this differs from the numpy default. See #5347`